src/scalaDebugger.ts

```diff
@@ -46,15 +50,161 @@ export async function start(
 
 class ScalaConfigProvider implements vscode.DebugConfigurationProvider {
   provideDebugConfigurations(): ProviderResult<DebugConfiguration[]> {
-    return [];
+    const mainClassPick = "Main Class";
+    const testClassPick = "Test Suite";
+
+    return vscode.window
+      .showQuickPick([mainClassPick, testClassPick], {
+        placeHolder: "Pick kind of class to debug",
+      })
+      .then((result) => {
+        if (result === mainClassPick) {
+          return this.provideDebugMainClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        } else if (result === testClassPick) {
+          return this.provideDebugTestClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        }
+        return [];
+      });
```

Comment:
```suggestion
      .then((result) => {
        switch (result) {
          case mainClassPick: return this.provideDebugMainClassConfiguration().then(config => [config]);
          case testClassPick: return this.provideDebugTestClassConfiguration().then(config => [config];
          default: return [];
        }
      }, () => [])
```

Reply:
The diff isn't very helpful, but I've simply moved the fallback to the outer `then` and replaced the if/else with a switch (which is a bit more readable IMO)

Reply:
(you'll need to run prettier, I think)

Reply:
done `switch`, thanks. To catch rejection, had to add another following `then`, `onrejection` here is not enough.